components/pedigree/resources/src/main/resources/pedigree/view.js

```diff
@@ -717,22 +717,24 @@ define([
 
             //timer.printSinceLast("=== highlight: ");
 
-            // re-evaluate which buttons & handles are appropriate for the nodes (e.g. twin button appears/disappears)
-            for (var nodeID in this._nodeMap) {
-                if (this._nodeMap.hasOwnProperty(nodeID)) {
-                    if (editor.getGraph().isPerson(nodeID) && !this.getNode(nodeID).getGraphics().getHoverBox().isMenuToggled()) {
-                        this.getNode(nodeID).getGraphics().getHoverBox().removeButtons();
-                        this.getNode(nodeID).getGraphics().getHoverBox().removeHandles();
+            if (!editor.isReadOnlyMode()) {
```

Comment:
This is "different" read-only mode. E.g. the user may have edit rights, but we disallow editing in old browsers, e.g. IE8 (since edit mode is harder to make compatible than view-only mode). Also, view-only mode can be used for pedigree preview on patient/family page (instead of a static SVG image), or to render template previews (since those are not editable, even in an editable pedigree)